### PR TITLE
SNOW-1852900:    Support Cortex function Sentiment in apply

### PR DESCRIPTION
**Pull request by sfc-gh-lmukhopadhyay**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-1852900

2. Fill out the following pre-review checklist:

   - [x] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.
   - [x] I acknowledge that I have ensured my changes to be thread-safe. Follow the link for more information: [Thread-safe Developer Guidelines](https://github.com/snowflakedb/snowpark-python/blob/main/CONTRIBUTING.md#thread-safe-development)

3. Please describe how your code solves the related issue.

   Support Cortex function Sentiment in apply.

### CODE REVIEW

**Review by sfc-gh-helmeleegy — COMMENTED** [review]
Can you also update `docs/source/snowpark/functions.rst`?

File: tests/integ/modin/test_apply_snowpark_python_functions.py [hunk 1]
```diff
@@ -89,3 +89,14 @@ def test_apply_snowflake_cortex_summarize():
...
+
+@sql_count_checker(query_count=1)
+@pytest.mark.skip("SNOW-1758914 snowflake.cortex.sentiment error on GCP")
+def test_apply_snowflake_cortex_sentiment():
+    from snowflake.snowpark.functions import snowflake_cortex_sentiment
+
+    content = "A very very bad review!"
+    s = pd.Series([content])
+    sentiment = s.apply(snowflake_cortex_sentiment).iloc[0]
+    assert -1 <= sentiment <= 1
```
Comment (sfc-gh-helmeleegy) [inline tests/integ/modin/test_apply_snowpark_python_functions.py:102]: Is it not safe to make the boundaries here between -1 and 0 instead of -1 and 1?

Comment (sfc-gh-helmeleegy) [reply tests/integ/modin/test_apply_snowpark_python_functions.py:102]: I realize that the return value may be slightly different each time we call the function. But I'd expect it to remain on the negative side for this particular review, no?

File: tests/integ/test_function.py [hunk 2]
```diff
@@ -2302,3 +2303,22 @@ def test_snowflake_cortex_summarize(session):
...
+
+    content = "A very very bad review!"
+    df = session.create_dataframe([[content]], schema=["content"])
+
+    sentiment_from_col = df.select(
+        snowflake_cortex_sentiment(col("content"))
+    ).collect()[0][0]
+    sentiment_from_str = df.select(snowflake_cortex_sentiment(content)).collect()[0][0]
+
+    assert -1 <= sentiment_from_col <= 1
```
Comment (sfc-gh-helmeleegy) [inline tests/integ/test_function.py:2323]: Same comment on the boundaries we're checking against.

File: tests/integ/modin/test_apply_snowpark_python_functions.py [hunk 3]
```diff
@@ -89,3 +89,14 @@ def test_apply_snowflake_cortex_summarize():
     summary = s.apply(snowflake_cortex_summarize).iloc[0]
     # this length check is to get around the fact that this function may not be deterministic
     assert 0 < len(summary) < len(content)
+
+
+@sql_count_checker(query_count=1)
+@pytest.mark.skip("SNOW-1758914 snowflake.cortex.sentiment error on GCP")
```
Comment (sfc-gh-azhan) [inline tests/integ/modin/test_apply_snowpark_python_functions.py:95]: Have you checked this?

File: tests/integ/modin/test_apply_snowpark_python_functions.py (hunk 1 above)
Comment (sfc-gh-lmukhopadhyay) [reply tests/integ/modin/test_apply_snowpark_python_functions.py:102]: That is true, if the cortex function works as expected this should be between -1 and 0. If we want to evaluate the validity of the cortex function itself, I can update this!

File: tests/integ/modin/test_apply_snowpark_python_functions.py (hunk 3 above)
Comment (sfc-gh-lmukhopadhyay) [reply tests/integ/modin/test_apply_snowpark_python_functions.py:95]: Yes, this test passes locally but fails the gcp precommit test here previously when the test was not skipped

**Review by sfc-gh-azhan — APPROVED** [review]

File: tests/integ/modin/test_apply_snowpark_python_functions.py (hunk 1 above)
Comment (sfc-gh-helmeleegy) [reply tests/integ/modin/test_apply_snowpark_python_functions.py:102]: Yes, I think it will make the test case more meaningful.

**Review by sfc-gh-helmeleegy — APPROVED** [review]
Approving, but please address the unresolved comments.

File: tests/integ/modin/test_apply_snowpark_python_functions.py (hunk 1 above)
Comment (sfc-gh-lmukhopadhyay) [reply tests/integ/modin/test_apply_snowpark_python_functions.py]: Sounds good, I've updated the sentiment tests!

File: tests/integ/test_function.py [hunk 4]
```diff
@@ -2302,3 +2303,23 @@ def test_snowflake_cortex_summarize(session):
...
     assert 0 < len(summary_from_str) < len(content)
+
+
+@pytest.mark.skipif(
+    "config.getoption('local_testing_mode', default=False)",
+    reason="FEAT: snowflake_cortex functions not supported",
+)
+def test_apply_snowflake_cortex_sentiment(session):
+    # TODO: SNOW-1758914 snowflake.cortex.sentiment error on GCP
+    if session.connection.host == "sfctest0.us-central1.gcp.snowflakecomputing.com":
```
Comment (sfc-gh-azhan) [inline tests/integ/test_function.py:2314]: we can use this in the tests/integ/modin/test_apply_snowpark_python_functions.py too

Comment (sfc-gh-lmukhopadhyay) [reply tests/integ/test_function.py:2314]: Should I also update `test_snowflake_cortex_summarize` in the meantime?

Comment (sfc-gh-azhan) [reply tests/integ/test_function.py:2317]: That will be helpful! You can either update in this PR or a new PR.

File: CHANGELOG.md [hunk 5]
```diff
@@ -2,13 +2,20 @@
...
+- Added support for `snowflake_cortex_sentiment`.
+
 ### Snowpark pandas API Updates
 
 #### New Features
 
 - Added support for `Series.str.ljust` and `Series.str.rjust`.
 - Added support for `Series.str.center`.
 - Added support for `Series.str.pad`.
+- Added support for applying Snowpark Python function `snowflake_cortex_sentiment`.
```
Comment (sfc-gh-yuwang) [inline CHANGELOG.md:18]:
```suggestion
- Added support for function `snowflake_cortex_sentiment` in `functions.py`.
```

File: src/snowflake/snowpark/functions.py [hunk 6]
```diff
@@ -10159,3 +10159,17 @@ def snowflake_cortex_summarize(text: ColumnOrLiteralStr):
...
     text_col = _to_col_if_lit(text, sql_func_name)
     return builtin(sql_func_name)(text_col)
+
+
+def snowflake_cortex_sentiment(text: ColumnOrLiteralStr):
+    """
+    A string containing the text for which a sentiment score should be calculated.
+    Args:
+        text: A string containing the English text from which a summary should be generated.
+    Returns:
```
Comment (sfc-gh-yuwang) [inline src/snowflake/snowpark/functions.py:10169]: Is it possible to add a simple example in the doc string?

Comment (sfc-gh-lmukhopadhyay) [reply src/snowflake/snowpark/functions.py:10169]: Sure, will add!

File: CHANGELOG.md (hunk 5 above)
Comment (sfc-gh-lmukhopadhyay) [reply CHANGELOG.md:18]: @sfc-gh-yuwang Did you mean to update the changelog for Snowpark Python API Updates or pandas API Updates? For pandas, we may want to keep the reference to `apply`

Comment (sfc-gh-aalam) [reply CHANGELOG.md:18]: I think the changelog for snowpark python api should be updated to what yuyang suggests here.

File: src/snowflake/snowpark/functions.py (hunk 6 above)
Comment (sfc-gh-aalam) [reply src/snowflake/snowpark/functions.py:10169]: +1

File: CHANGELOG.md (hunk 5 above)
Comment (sfc-gh-lmukhopadhyay) [reply CHANGELOG.md:18]: Makes sense, it should be updated now!

**Review by sfc-gh-yuwang — APPROVED** [review]

**Pull request merged**